lib/imagefs.coffee

```diff
@@ -86,11 +87,29 @@ read = (disk, partition, path) ->
 	composeDisposers(
 		driver.interact(disk, partition)
 		(fs_) ->
-			Promise.resolve(fs_.createReadStream(path, autoClose: false))
+			readStream = null
+			outputStream = new stream.PassThrough()
+
+			# We don't start the stream until somebody else starts listening
+			startReadStream = ->
+				outputStream.removeListener('newListener', startReadStream)
+
+				# This is triggered _before_ the listener is added, so wait briefly
+				process.nextTick ->
+					readStream = fs_.createReadStream(path, autoClose: false)
+					readStream.pipe(outputStream)
+					readStream.on 'error', (err) ->
```

Comment:
Can you set up the error listener before the pipe?

Reply:
Sure, done